src/init.ts

```diff
@@ -0,0 +1,85 @@
+import { DEFAULT_DOMAIN, JAAS_DOMAIN } from './constants';
+import IJitsiMeetExternalApi from './types/IJitsiMeetExternalApi';
+
+type ExternalApi = {
+    isLoaded: boolean;
+    callbacks: ((err: Error | null, api?: new () => IJitsiMeetExternalApi) => void)[];
+    err: Error | null;
+};
+
+let externalApi: ExternalApi;
+
+export const initExternalApi = (
```

Comment:
I still find this more complex than it needs to be. Maybe we can borrow some inspiration from https://github.com/feross/load-script2 ?

I think we can just have some state booleans to tell us if we are loading the external API, in which case we could chain a promise, or if it's loaded already, etc.

Can you give it another go?